docs/src/keyboard_configuration.md

```diff
@@ -228,11 +245,25 @@ usb_enable = true
 
 # Set matrix IO for the board. This section is for non-split keyboard and is conflict with [split] section
 [matrix]
+# `matrix_type` is optional. Default is "normal"
+matrix_type = "normal"
 # Input and output pins
 input_pins = ["PIN_6", "PIN_7", "PIN_8", "PIN_9"]
 output_pins = ["PIN_19", "PIN_20", "PIN_21"]
 # WARNING: Currently row2col/col2row is set in RMK's feature gate, configs here do nothing actually
 
+# Direct Pin Matrix is a Matrix of buttons connected directly to pins. It conflicts with the above.
+matrix_type = "direct_pin"
+direct_pins = [
```

Comment:
This type of direct pin matrix seems invalid now:


```toml
direct_pins = [
    ["PIN_0", "PIN_1", "PIN_2"],
    ["PIN_3", "_", "PIN_5"]
]
```

It's common that the key matrix doesn't have keys at all positions.
Normal pin matrix doesn't have this issue, since the scanning algorithm automatically ignore those positions. But for direct pin matrix, we should allow users ignore some positions in the matrix(even though users can still fill a dummy pin here to bypass)

Reply:
Done! [feat(direct_pin, macro): support no_pin in matrix](https://github.com/HaoboGu/rmk/pull/116/commits/adde5e868262604c1983d1ec488c54eea2a95af5)